services/cirrus/cirrus.service.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+const Joi = require('joi')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  message: Joi.alternatives()
+    .try(isBuildStatus, Joi.equal('unknown'))
+    .required(),
+}).required()
+
+module.exports = class Cirrus extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'cirrus',
+      pattern: '/github/:user/:repo',
+    }
+  }
```

Comment:
Any additional parameters that you want to expose to end users as query parameters have to be explicitly added to the route in order to be correctly handled. 

Check the [CircleCI badge route](https://github.com/badges/shields/blob/master/services/circleci/circleci.service.js#L27) for a live reference, as well as the inline notes/docs [here](https://github.com/badges/shields/blob/master/core/base-service/base.js#L93) for defining a schema for the query params. 

Additionally, note that `branch` should be added as an optional route pattern (`:branch*`) and not a query param. This is more of a special case due to the way `branch` has historically been handled, so we'll need to keep it as a route param here too (i.e. `/github/:user/:repo/:branch*`)

Lastly, if Cirrus supports other VCS Hosts beyond GitHub (GitLab, Bitbucket, etc.) then we need to make sure that's available to users via the route param (the Circle reference is applicable here as well).

Reply:
It only supports GH right now

Reply:
Gotcha thanks. Suggested changes added inline above to add query param schema and update route accordingly